src/main/java/com/webcrawling/service/webCrawler/impl/WebCrawlerServiceImpl.java

```diff
@@ -31,8 +32,10 @@ public class WebCrawlerServiceImpl implements WebCrawlerService {
     WebPageRepository webPageRepository;
     @Autowired
     private ExceptionRecordRepository exceptionRecordRepository;
-        public void  crawlWebsite(String startingUrl, int maxDepth, int maxDocuments) {
+    @Async
+    public void  crawlWebsite(String startingUrl, int maxDepth, int maxDocuments) {
```

Comment:
@Async worked in different way Like if you have an endpoint and you are going to process the huge amount of data and user should not wait for that endpoint it expect response in seconds and your processing start in background.
you can find details in this blog 
https://medium.com/javarevisited/asynchronous-calls-in-spring-boot-using-async-6846219c4ab1